server/app.ts

```diff
@@ -1,73 +1,75 @@
-import { topRatedMailJob } from './notifications/notification.jobs';
+import * as cookieParser from 'cookie-parser'
+import * as cors from 'cors'
+import * as express from 'express'
+import * as helmet from 'helmet'
+import * as hpp from 'hpp'
+import * as logger from 'morgan'
+import Routes from './core/route.interface'
+import errorMiddleware from './core/middleware/error.middleware'
+import { database } from './core/database'
+import pageViewMiddleware from './core/middleware/pageview.middleware'
+import { topRatedMailJob } from './notifications/notification.jobs'
+import { purgeAllCachesJob } from './core/cache/cache.job'
 
 class App {
-	public app: express.Application;
-	public port: (string | number);
-	public env: boolean;
+	public app: express.Application
+	public port: (string | number)
+	public env: boolean
 
 	constructor(routes: Routes[]) {
-		this.app = express();
-		this.port = process.env.PORT || 3000;
-		this.env = process.env.NODE_ENV === 'production' ? true : false;
+		this.app = express()
+		this.port = process.env.PORT || 3000
+		this.env = process.env.NODE_ENV === 'production' ? true : false
 
-		this.initializeMiddlewares();
-		this.initializeRoutes(routes);
-		this.initializeErrorHandling();
-		this.initializeScheduledJobs();
+		this.initializeMiddlewares()
+		this.initializeRoutes(routes)
+		this.initializeErrorHandling()
+		this.initializeScheduledJobs()
 	}
 
 	public listen() {
 		this.app.listen(this.port, () => {
-			console.log(`🚀 App listening on the port ${this.port}`);
-		});
+			console.log(`🚀 App listening on the port ${this.port}`)
+		})
 	}
 
 	public getServer() {
-		return this.app;
+		return this.app
 	}
 
 	private initializeMiddlewares() {
 		if (this.env) {
-			this.app.use(hpp());
-			this.app.use(helmet());
-			this.app.use(logger('combined'));
-			this.app.use(cors({ origin: process.env.DOMAIN, credentials: true }));
+			this.app.use(hpp())
+			this.app.use(helmet())
+			this.app.use(logger('combined'))
+			this.app.use(cors({ origin: process.env.DOMAIN, credentials: true }))
 		} else {
-			this.app.use(logger('dev'));
-			this.app.use(cors({ origin: true, credentials: true }));
+			this.app.use(logger('dev'))
+			this.app.use(cors({ origin: true, credentials: true }))
 		}
-		database.sync({ force: false });
-		this.app.use(express.json());
-		this.app.use(express.urlencoded({ extended: true }));
-		this.app.use(cookieParser());
+		database.sync({ force: false })
+		this.app.use(express.json())
+		this.app.use(express.urlencoded({ extended: true }))
+		this.app.use(cookieParser())
 		this.app.use(pageViewMiddleware())
 	}
 
 	private initializeRoutes(routes: Routes[]) {
 		routes.forEach((route) => {
-			this.app.use('/', route.router);
-		});
+			this.app.use('/', route.router)
+		})
 	}
 
 	private initializeErrorHandling() {
-		this.app.use(errorMiddleware);
+		this.app.use(errorMiddleware)
 	}
 
 	private initializeScheduledJobs() {
-		topRatedMailJob();
+		topRatedMailJob()
```

Comment:
It might be a good idea to "inject" cron expressions here, so we have them all visible in the same place, to ensure that we have no two jobs running at the same time, potentially causing race conditions.

We could even wrap cron expressions in some nice builder API, and use it like this: 
typescript```
topRatedMailJob(Cron.everyHourAtMinute(50))
purgeAllCachesJob(Cron.everyDayAtHours(6, 12))
```